Tests/CreateSliceForDiagnostics/main.cpp

```diff
@@ -0,0 +1,371 @@
+
+       if(slicetypeToBeConverted==1) {
+          cs_mf[0].reset(new MultiFab(amrex::convert(crse_ba[0],convertTypeofSlice),sdmap[0],ncomp,nghost));
+       }
+       else {
+          cs_mf[0].reset(new MultiFab(crse_ba[0],sdmap[0],ncomp,nghost));
+       }
+
+       // Assumption :: Currently works only for 1 level //
+       for (int ilev = 0; ilev < nlevs; ilev++)
+       {
+           MultiFab& mfSrc = *smf[ilev];
+           MultiFab& mfDst = *cs_mf[ilev];
+          
+           MFIter mfi_dst(mfDst);
+           for (MFIter mfi(mfSrc); mfi.isValid(); ++mfi)
+           {
+               FArrayBox& Src_fabox = mfSrc[mfi];
+
+               const Box& Dst_bx = mfi_dst.validbox();
+               FArrayBox& Dst_fabox = mfDst[mfi_dst];
+
+               int scomp = 0;
+               int dcomp = 0;
+
+               if( convertTypeofSlice==cctype ) {
+                  amrex::amrex_avgdown(Dst_bx,Dst_fabox,Src_fabox,dcomp,scomp,ncomp,cr_ratio);
+               }
+               if( convertTypeofSlice == ndtype ) {
+                  amrex::amrex_avgdown_nodes(Dst_bx,Dst_fabox,Src_fabox,dcomp,scomp,ncomp,cr_ratio);
+               }
+               if( convertTypeofSlice == xetype ) {
+                  amrex::amrex_avgdown_edges(Dst_bx,Dst_fabox,Src_fabox,dcomp,scomp,ncomp,cr_ratio,0);
+               }
+               if( convertTypeofSlice == yetype) {
+                  amrex::amrex_avgdown_edges(Dst_bx,Dst_fabox,Src_fabox,dcomp,scomp,ncomp,cr_ratio,1);
+               }
+               if( convertTypeofSlice == zetype ) {
+                  amrex::amrex_avgdown_edges(Dst_bx,Dst_fabox,Src_fabox,dcomp,scomp,ncomp,cr_ratio,2);
+               }
+               if (AMREX_SPACEDIM==3) {
+                  if( convertTypeofSlice == xftype) {
+                     amrex::amrex_avgdown_faces(Dst_bx,Dst_fabox,Src_fabox,dcomp,scomp,ncomp,cr_ratio,0);
+                  }
+                  if( convertTypeofSlice == yftype ) {
+                     amrex::amrex_avgdown_faces(Dst_bx,Dst_fabox,Src_fabox,dcomp,scomp,ncomp,cr_ratio,1);
+                  }
+                  if( convertTypeofSlice == zftype ) {
+                     amrex::amrex_avgdown_faces(Dst_bx,Dst_fabox,Src_fabox,dcomp,scomp,ncomp,cr_ratio,2);
+                  }
+               }
+
+               if ( mfi_dst.isValid() ) {
+                  ++mfi_dst;
+               }
+           }
+       }
+       VisMF::Write((*cs_mf[0]),"vismf_output");
+    }
+
+    }
+}
+
+// assuming 6 components for all multifabs just for testing purposes //
+void InitializeVariables(Vector<MultiFab*> mf, Vector<Geometry> geom)
+{
+    int nlevs = geom.size();
+    const Real *dom_dx = geom[0].CellSize();
+
+    for (int lev = 0; lev < nlevs; ++lev) {
+        for (MFIter mfi(*mf[lev]); mfi.isValid(); ++mfi){
+            const Box& bx = mfi.validbox();
+            FArrayBox& fbox_Dst = (*mf[lev])[mfi];
+            for (IntVect cell=bx.smallEnd(); cell<=bx.bigEnd(); bx.next(cell)) 
```

Comment:
It doesn't really matter here, but this style of looping over the IntVects in a Box has a fair bit of overhead. In general, for looping over the cells in a FAB and doing stuff in C++, it's better to use the style in the `AMReX_MultiFabUtil_3D.C`, except using the `array` function instead of `view`.